pdm/pep517/editable.py

```diff
@@ -0,0 +1,123 @@
+import hashlib
+import os
+import zipfile
+from base64 import urlsafe_b64encode
+from pathlib import Path
+from typing import Any, Dict, Iterable, List, Mapping, Optional, Tuple, Union
+
+from pdm.pep517.base import BuildError
+from pdm.pep517.wheel import WheelBuilder
+
+
+class EditableProject:
+    """Copied from https://github.com/pfmoore/editables"""
```

Comment:
Any reason you're copying rather than depending on editables? The `map` functionality means you need a runtime dependency on editables anyway, so is there any reason for not also using it here?

Reply:
If i understand your review correctly, first this project is vendoring deps to keep it zero-dependency and i just don't want to break it. Second, this small class is not many code to maintain. And the last, runtime dependency has nothing to do with build time.

Reply:
No worries, I just wanted to make sure you'd understood that you still needed the runtime dependency, and that there wasn't any clarification I needed to add in the docs.

I'm glad the library was useful for you :-)